etc/kernel-launchers/kubernetes/scripts/launch_kubernetes.py

```diff
@@ -29,33 +46,10 @@ def launch_kubernetes_kernel(kernel_id, response_addr, spark_context_init_mode):
     # with name in lower case.
     for name, value in os.environ.items():
         if name.startswith('KERNEL_'):
-            keywords[name.lower()] = value
-
-    # Read the kernel-pod yaml file, stripping off any commented lines.  This allows instances of the
-    # yaml file to comment out substitution parameters since we want to fail the launch if any are left
-    # unsubstituted.  Otherwise, commented out parameters could fail the launch if they had no substitutions.
-    #
-    yaml_template = ''
-    with open(os.path.join(os.path.dirname(__file__), "kernel-pod.yaml")) as f:
-        for line in f:
-            line = line.split('#', 1)[0]
-            yaml_template = yaml_template + line
-        f.close()
-
-    # Perform substitutions, then verify all parameters have been replaced.  If any
-    # parameters still exist, print their names and exit.  If all have been replaced,
-    # iterate over each document, issue creation statements.
-    #
-    k8s_yaml = Template(yaml_template).safe_substitute(keywords)
+            keywords[name.lower()] = yaml.safe_load(value)
 
-    # Check for non-substituted parameters - exit if found.
-    #
-    missing_params = [param[1] for param in Formatter().parse(k8s_yaml) if param[1]]
-    if len(missing_params) > 0:
-        missing_params = ['${' + param[1] + '}' for param in Formatter().parse(k8s_yaml) if param[1]]
-        if len(missing_params) > 0:
-            sys.exit("ERROR - The following parameters were not substituted - kernel launch terminating! {}".
-                     format(missing_params))
+    # Substitute all template variable (wrapped with {{ }}) and generate `yaml` string.
+    k8s_yaml = generate_kernel_pod_yaml(keywords)
```

Comment:
Nice - this will fit in well when we (can) support [Spark POD templates](https://github.com/jupyter/enterprise_gateway/pull/559) in Spark 3.0.